pallets/did/src/did_details.rs

```diff
@@ -634,32 +635,57 @@ pub struct DidUpdateDetails<T: Config> {
 	/// A new set of key agreement keys to add to the ones already stored.
 	pub new_key_agreement_keys: BTreeSet<DidEncryptionKey>,
 	/// \[OPTIONAL\] The attestation key update action.
-	pub attestation_key_update: DidVerificationKeyUpdateAction,
+	pub attestation_key_update: DidFragmentUpdateAction<DidVerificationKey>,
 	/// \[OPTIONAL\] The delegation key update action.
-	pub delegation_key_update: DidVerificationKeyUpdateAction,
+	pub delegation_key_update: DidFragmentUpdateAction<DidVerificationKey>,
 	/// The set of old attestation keys to remove, given their identifiers.
 	/// If the operation also replaces the current attestation key, it will
 	/// not be considered for removal in this operation, so it is not
 	/// possible to specify it for removal in this set.
 	pub public_keys_to_remove: BTreeSet<KeyIdOf<T>>,
-	/// \[OPTIONAL\] The new endpoint URL.
-	pub new_endpoint_url: Option<Url>,
+	/// The update action on the service endpoints information.
+	pub service_endpoints_update: DidFragmentUpdateAction<ServiceEndpoints>,
 }
 
-/// Possible actions on a DID verification key within a
-/// [DidUpdateOperation].
-#[derive(Clone, Decode, Debug, Encode, Eq, Ord, PartialEq, PartialOrd)]
-pub enum DidVerificationKeyUpdateAction {
-	/// Do not change the verification key.
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq)]
+pub struct ServiceEndpoints {
+	pub content_hash: Hash,
+	pub urls: Vec<Url>,
+	pub content_type: ContentType,
+}
+
+impl ServiceEndpoints {
+	pub(crate) fn validate_against_config_limits<T: Config>(&self) -> Result<(), InputError> {
```

Comment:
Being super picky like I am: would be fine to call it just validate right?

Reply:
Mhh yeah, but validate name is so generic that you don't really know what it does without taking a look inside, right?

Reply:
it validates! 😁 What exactly is validated should be written in the doc string or you have to look in code IMHO.

Reply:
Once again, I tend to disagree, especially because it takes `Config` as a generic 😄 So I'll leave it as it is for now, but we can have a discussion once I'll be back 🏝

Reply:
>  So I'll leave it as it is for now, but we can have a discussion once I'll be back 🏝

Shouldn't we merge this before you are back for the audit?

Reply:
Yes, I'm fixing the migration stuff and I'll push soon. Then you guys can merge when all the tests pass again.

Reply:
no need to discuss this. If you like it more this way, leave it as it is.